eth/db/header.py

```diff
@@ -149,43 +161,60 @@ def header_exists(self, block_hash: Hash32) -> bool:
     def persist_header(self,
                        header: BlockHeader
                        ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
+        return self.persist_header_chain((header,))
+
+    def persist_header_chain(self,
+                             headers: Iterable[BlockHeader]
+                             ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
         """
-        :returns: iterable of headers newly on the canonical chain
+        Return two iterable of headers, the first containing the new canonical headers,
+        the second containing the old canonical headers
         """
-        if header.parent_hash != GENESIS_PARENT_HASH:
-            try:
-                self.get_block_header_by_hash(header.parent_hash)
-            except HeaderNotFound:
+
+        try:
+            first_header = first(headers)
+        except StopIteration:
```

Comment:
I guess this could be avoided if we used a `Tuple` instead of accepting all `Iterable` but then, it strikes me as odd to restrict to a concrete type where an abstraction would do it.

Reply:
Yep, I just have a hard time accepting concrete types as params to an API if the API actually doesn't need that concrete type and could rather work with an abstraction. Another alternative would be to use `Sequence[BlockHeader]` as that allows for a simple `len()` check. But if you are fine with the `try` / `except` I' fine, too. :)

Reply:
Personally, I think it's kinda cool to be able to something like:

```py
chaindb.persist_header_chain(reversed(backwards_headers))
```

which wouldn't work with `Sequence`.